### DISCUSSION TITLE
Execute command when a profile is open

### DISCUSSION THREAD

**Discussion by Zediz** [microsoft/terminal#14791] [Q&A] [answered]
It would be cool if you can add a option on each profile. That allow us to run a command or several commands when the profile is opening. 

For example a basic usage could be, at open a angular project profile, run ng serve when the profile is selected. or someting similar.

**Comment by zadjii-msft** [comment, marked as answer]
You mean like a [`commandline`](https://learn.microsoft.com/en-us/windows/terminal/customize-settings/profile-general#command-line)?

**Reply by Zediz** [reply]
Yes, It would be helpfull, I saw this feature on iTerm2. 
You click to open a custom profile, and then it opens and at the first line it runs the command your previusly set... :)
